wren-ai-service/Makefile

```diff
@@ -1,61 +1,243 @@
-start:
-	poetry run python -m src.__main__
+# Variables
+WREN_AI_SERVICE_PORT=5556
 
-build:
-	docker compose -f docker/docker-compose.yml --env-file .env.prod build
+# ANSI color codes
+GREEN=\033[0;32m
+YELLOW=\033[0;33m
+RED=\033[0;31m
+BLUE=\033[0;34m
+RESET=\033[0m
 
-up:
-	make run-wren-engine && \
-	docker compose -f docker/docker-compose.yml --env-file .env.prod up -d
+setup:
+	@echo "$(YELLOW)Setting up development environment...$(RESET)"
+	@$(MAKE) check-python
+	@$(MAKE) check-poetry
+	@$(MAKE) install-python-dependencies
+	@$(MAKE) install-precommit-hooks
+	@$(MAKE) setup-envfile
+	@echo "$(GREEN)Development environment setup complete!$(RESET)"
 
-down:
-	make stop-wren-engine && \
-	docker compose -f docker/docker-compose.yml --env-file .env.prod down
+check-python:
```

Comment:
as we said in requirements in README, we need python 3.12. so can we also check the python version should be 12.xxx here?